src/main/java/edu/harvard/iq/dataverse/api/BuiltinUsers.java

```diff
@@ -122,6 +128,16 @@ public Response create(BuiltinUser user, @PathParam("password") String password,
     public Response createWithNotification(BuiltinUser user, @PathParam("password") String password, @PathParam("key") String key, @PathParam("sendEmailNotification") Boolean sendEmailNotification) {
         return internalSave(user, password, key, sendEmailNotification);
     }
+
+    @GET
+    @Path("{username}/canLoginWithGivenCredentials")
```

Comment:
Is this secure enough? I believe it is blocked by the default :BlockedApiEndpoints setting and restricted to localhost by default, but it is a way to scan passwords. Should it have its own Key, like the [BuiltinUsers.KEY](https://guides.dataverse.org/en/latest/installation/config.html#id377), or a throttle? Would it be safer to copy the validation logic from the authenticationService into the storage provider and just query the db (straight-forward except for the rarely/never used algorithm update code?)?

Separate q: Unless I missed it, this doesn't check for a user being deactivated - is that what's desired? I think w/o that you'd get logged in at Keycloak, but would then get errors when other API calls are made for the user? Should login just be denied here?